backend/capellacollab/core/authentication/injectables.py

```diff
@@ -150,36 +150,16 @@ def __call__(
         username: str = fastapi.Depends(get_username),
         db: orm.Session = fastapi.Depends(database.get_db),
     ) -> bool:
-        if not (user := users_crud.get_user_by_name(db, username)):
-            if self.verify:
-                raise fastapi.HTTPException(
-                    status_code=status.HTTP_404_NOT_FOUND,
-                    detail={"reason": f"User {username} was not found"},
-                )
+        if not (user := self._get_user_and_check(username, db)):
```

Comment:
This seems to be a bit overkill for me and doesn't reduce the complexity in the code as intended by introducing McCabe. 

Instead, I'd propose do change it properly and update to our new exception style. Instead of `fastapi.HTTPException`, raise custom exceptions. Those exceptions all inherit from a general exception class. 

We can then remove the `verify` option. All places where `verify=False` was used have to be refactored to catch the error with try & except.

Reply:
See also: https://github.com/DSD-DBS/capella-collab-manager/issues/945

Reply:
Dominik and I have agreed on only changing the fastapi.HTTPExceptions to BaseExceptions.

Reply:
I'll do the definition of exceptions as part of https://github.com/DSD-DBS/capella-collab-manager/pull/1569.